### ISSUE TITLE
yfinance error when running 'streamlit run main.py' 

### ISSUE THREAD

**Issue by josuebatista** [mrspiggot/FinGPT#1]
There is an issue with yfinance. As soon as I execute 'steamlit run main.py' I get the following error:

HTTPError: 401 Client Error: Unauthorized for url: https://query2.finance.yahoo.com/v10/finance/quoteSummary/AAPL?modules=summaryProfile%2CfinancialData%2CquoteType%2CdefaultKeyStatistics%2CassetProfile%2CsummaryDetail&ssl=true
Traceback:
File "/home/josue/miniconda3/envs/fingpt/lib/python3.8/site-packages/streamlit/runtime/scriptrunner/script_runner.py", line 565, in _run_script
    exec(code, module.__dict__)
File "/home/josue/FinGPT/main.py", line 96, in <module>
    long_description = ticker.info["longBusinessSummary"]
File "/home/josue/miniconda3/envs/fingpt/lib/python3.8/site-packages/yfinance/ticker.py", line 138, in info
    return self.get_info()
File "/home/josue/miniconda3/envs/fingpt/lib/python3.8/site-packages/yfinance/base.py", line 1020, in get_info
    data = self._quote.info
File "/home/josue/miniconda3/envs/fingpt/lib/python3.8/site-packages/yfinance/scrapers/quote.py", line 555, in info
    self._fetch(self.proxy)
File "/home/josue/miniconda3/envs/fingpt/lib/python3.8/site-packages/yfinance/scrapers/quote.py", line 706, in _fetch
    result = self._data.get_raw_json(
File "/home/josue/miniconda3/envs/fingpt/lib/python3.8/site-packages/yfinance/data.py", line 209, in get_raw_json
    response.raise_for_status()
File "/home/josue/miniconda3/envs/fingpt/lib/python3.8/site-packages/requests/models.py", line 1021, in raise_for_status
    raise HTTPError(http_error_msg, response=self)

**Comment by mrspiggot** [comment]
I just cloned the repo into a clean environment on my Mac, ran pip install -r requirement.txt and imported a .env file. Everything ran fine.I then tried it under a conda install and ran into the same issues you describe.Whether the mismatch between conda and the app originates in my code, or in yfinance, or somewhere else I can’t say. But there does seem to be an incompatibility somewhere.  If I have time later this week I might take a look at why this is, but clearly this might not be a quick fix.  My advice, if possible, is to ignore conda and use pip. If that is not possible the. I’ll try and find out why - but Be warned! I’m an AI guy, and not an expert in the nuances of python/pip/conda! So it might take a while, and even then I might not figure out where the incompatibility lies.But a huge thank-you for bringing this to my attention.At the very least I need to let people know that there seems to be an issue with people running the code under conda (a while back in the Discord another member reported some trouble using conda too).RichardOn 15 Jul 2023, at 03:42, Josue R Batista ***@***.***> wrote:﻿
There is an issue with yfinance. As soon as I execute 'steamlit run main.py' I get the following error:
HTTPError: 401 Client Error: Unauthorized for url: https://query2.finance.yahoo.com/v10/finance/quoteSummary/AAPL?modules=summaryProfile%2CfinancialData%2CquoteType%2CdefaultKeyStatistics%2CassetProfile%2CsummaryDetail&ssl=true
Traceback:
File "/home/josue/miniconda3/envs/fingpt/lib/python3.8/site-packages/streamlit/runtime/scriptrunner/script_runner.py", line 565, in _run_script
exec(code, module.dict)
File "/home/josue/FinGPT/main.py", line 96, in 
long_description = ticker.info["longBusinessSummary"]
File "/home/josue/miniconda3/envs/fingpt/lib/python3.8/site-packages/yfinance/ticker.py", line 138, in info
return self.get_info()
File "/home/josue/miniconda3/envs/fingpt/lib/python3.8/site-packages/yfinance/base.py", line 1020, in get_info
data = self._quote.info
File "/home/josue/miniconda3/envs/fingpt/lib/python3.8/site-packages/yfinance/scrapers/quote.py", line 555, in info
self._fetch(self.proxy)
File "/home/josue/miniconda3/envs/fingpt/lib/python3.8/site-packages/yfinance/scrapers/quote.py", line 706, in _fetch
result = self._data.get_raw_json(
File "/home/josue/miniconda3/envs/fingpt/lib/python3.8/site-packages/yfinance/data.py", line 209, in get_raw_json
response.raise_for_status()
File "/home/josue/miniconda3/envs/fingpt/lib/python3.8/site-packages/requests/models.py", line 1021, in raise_for_status
raise HTTPError(http_error_msg, response=self)

**Comment by josuebatista** [comment]
Hi Richard, I am also looking into the source of the issue and resolution. It might have something to do on how the conda environment manages crumbs and cookies, or the version of yfinance (i.e. V10 vs. V7 or V6). I'll keep you posted on my progress.

**Comment by mrspiggot** [comment]
Thanks for the update, and thanks for taking the time to look into it. Greatly appreciated!RichardOn 15 Jul 2023, at 14:53, Josue R Batista ***@***.***> wrote:﻿
Hi Richard, I am also looking into the source of the issue and resolution. It might have something to do on how the conda environment manages crumbs and cookies, or the version of yfinance (i.e. V10 vs. V7 or V6). I'll keep you posted on my progress.

**Comment by josuebatista** [comment]
Hi Richard, I spent a few hours looking into the source of the authorization error. First, I found that the latest version of yfinance==0.2.24 does not trigger the 'unauthorized, invalid crumb' error. That is because a code fix to the yfinance/scrapers/quote.py file was committed a few days ago. It fixes the error: unauthorized, invalid crumb.

https://github.com/ranaroussi/yfinance/commit/509a109f29a657190e7af86a160db340d1dff71d

There is a catch in the yfinance==0.2.24 version: yfinance/scrapers/fundamentals.py has not yet implemented the functions earnings() and shares(), which throws errors when streamlit calls FinGPT/main.py.

Thus, I regressed to yfinance==0.2.18 as per the FinGPT/requirements.txt; then I manually applied the fix to the yfinance/scrapers/quote.py, which worked well and eliminated the unauthorized, invalid crumb errors. However, I experienced another error:
`
Traceback (most recent call last):
  File "/home/josue/miniconda3/envs/fingpt/lib/python3.8/site-packages/streamlit/runtime/scriptrunner/script_runner.py", line 565, in _run_script
    exec(code, module.__dict__)
  File "/home/josue/FinGPT/main.py", line 96, in <module>
    long_description = ticker.info["longBusinessSummary"]
**KeyError: 'longBusinessSummary'**
`

Essentially, ticker.info["longBusinessSummary"] does not exist in yfinance==0.2.18:

`
(fingpt) josue@OptiPlex-9020-01:~/FinGPT$ python
Python 3.8.17 (default, Jul  5 2023, 21:04:15) 
[GCC 11.2.0] :: Anaconda, Inc. on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import pkg_resources
>>> yfinance_version = pkg_resources.get_distribution("yfinance").version
>>> print(yfinance_version)
0.2.18
>>> import yfinance as yf
>>> ticker_yq = yf.Ticker('goog')
>>> info = ticker_yq.info
>>> info.keys()
dict_keys(['maxAge', 'priceHint', 'previousClose', 'open', 'dayLow', 'dayHigh', 'regularMarketPreviousClose', 'regularMarketOpen', 'regularMarketDayLow', 'regularMarketDayHigh', 'payoutRatio', 'beta', 'trailingPE', 'forwardPE', 'volume', 'regularMarketVolume', 'averageVolume', 'averageVolume10days', 'averageDailyVolume10Day', 'bid', 'ask', 'bidSize', 'askSize', 'marketCap', 'fiftyTwoWeekLow', 'fiftyTwoWeekHigh', 'priceToSalesTrailing12Months', 'fiftyDayAverage', 'twoHundredDayAverage', 'trailingAnnualDividendRate', 'trailingAnnualDividendYield', 'currency', 'trailingPegRatio'])
`


However, it does exist in yfinance==0.2.24

`(fingpt) josue@OptiPlex-9020-01:~/FinGPT$ python
Python 3.8.17 (default, Jul  5 2023, 21:04:15) 
[GCC 11.2.0] :: Anaconda, Inc. on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import pkg_resources
>>> yfinance_version = pkg_resources.get_distribution("yfinance").version
>>> print(yfinance_version)
0.2.24
>>> import yfinance as yf
>>> ticker_yq = yf.Ticker('goog')
>>> info = ticker_yq.info
>>> info.keys()
dict_keys(['address1', 'city', 'state', 'zip', 'country', 'phone', 'website', 'industry', 'industryDisp', 'sector', **'longBusinessSummary'**, 'fullTimeEmployees', 'companyOfficers', 'auditRisk', 'boardRisk', 'compensationRisk', 'shareHolderRightsRisk', 'overallRisk', 'governanceEpochDate', 'compensationAsOfEpochDate', 'maxAge', 'priceHint', 'previousClose', 'open', 'dayLow', 'dayHigh', 'regularMarketPreviousClose', 'regularMarketOpen', 'regularMarketDayLow', 'regularMarketDayHigh', 'payoutRatio', 'beta', 'trailingPE', 'forwardPE', 'volume', 'regularMarketVolume', 'averageVolume', 'averageVolume10days', 'averageDailyVolume10Day', 'bid', 'ask', 'bidSize', 'askSize', 'marketCap', 'fiftyTwoWeekLow', 'fiftyTwoWeekHigh', 'priceToSalesTrailing12Months', 'fiftyDayAverage', 'twoHundredDayAverage', 'trailingAnnualDividendRate', 'trailingAnnualDividendYield', 'currency', 'enterpriseValue', 'profitMargins', 'floatShares', 'sharesOutstanding', 'sharesShort', 'sharesShortPriorMonth', 'sharesShortPreviousMonthDate', 'dateShortInterest', 'sharesPercentSharesOut', 'heldPercentInsiders', 'heldPercentInstitutions', 'shortRatio', 'impliedSharesOutstanding', 'bookValue', 'priceToBook', 'lastFiscalYearEnd', 'nextFiscalYearEnd', 'mostRecentQuarter', 'earningsQuarterlyGrowth', 'netIncomeToCommon', 'trailingEps', 'forwardEps', 'pegRatio', 'lastSplitFactor', 'lastSplitDate', 'enterpriseToRevenue', 'enterpriseToEbitda', '52WeekChange', 'SandP52WeekChange', 'exchange', 'quoteType', 'symbol', 'underlyingSymbol', 'shortName', 'longName', 'firstTradeDateEpochUtc', 'timeZoneFullName', 'timeZoneShortName', 'uuid', 'messageBoardId', 'gmtOffSetMilliseconds', 'currentPrice', 'targetHighPrice', 'targetLowPrice', 'targetMeanPrice', 'targetMedianPrice', 'recommendationMean', 'recommendationKey', 'numberOfAnalystOpinions', 'totalCash', 'totalCashPerShare', 'ebitda', 'totalDebt', 'quickRatio', 'currentRatio', 'totalRevenue', 'debtToEquity', 'revenuePerShare', 'returnOnAssets', 'returnOnEquity', 'grossProfits', 'freeCashflow', 'operatingCashflow', 'earningsGrowth', 'revenueGrowth', 'grossMargins', 'ebitdaMargins', 'operatingMargins', 'financialCurrency', 'trailingPegRatio'])
>>> `

I am curious, what version of yfinance are you using in your latest clean install? Also, what OS did you do the clean install on, windows, linux, macOS? Did you use a python environment manager, i.e. virtualenv, pipenv, pyenv?
Thank you!
